### PR TITLE
NMS-10243:  apply custom tag to nsc-alarms.jsp

### PR DESCRIPTION
**Pull request by patrick-schweizer**
* JIRA: https://issues.opennms.org/browse/NMS-10243

Our [continuous integration system](http://bamboo.internal.opennms.com:8085) will test and verify your changes.

### CODE REVIEW

**Comment by j-white** [comment]
Looks reasonable.

**Comment by patrick-schweizer** [comment]
@j-white is the location of the new module opennms-taglib correct?

**Comment by j-white** [comment]
We tend not to create many top level projects anymore, and root them under `core` or `features`, but I don't have any issues with the current structure.

**Comment by patrick-schweizer** [comment]
The PR fails because it can't find the artifact that is introduced by this PR (from https://bamboo.opennms.org/browse/OPENNMS-ONMS2306-4/log):

build	08-Jul-2018 19:35:33	[ERROR] Failed to execute goal on project opennms-webapp: Could not resolve dependencies for project org.opennms:opennms-webapp:war:21.1.0-SNAPSHOT: Could not find artifact org.opennms:opennms-taglib:jar:21.1.0-SNAPSHOT -> [Help 1]

@Anyone: ideas what to do? Thanks!

**Comment by mvrueden** [comment]
At the moment all tags are defined in `opennms-webapp`, see: https://github.com/OpenNMS/opennms/blob/develop/opennms-webapp/src/main/webapp/WEB-INF/taglib.tld 
I'd prefer to keep that structure instead of introducing a new one, unless it is necessary.

**Comment by mvrueden** [comment]
@patrick-schweizer it does not compile, because `opennms-taglib` is not defined as a `module` in the root pom and the `tld` file was defined in module `opnnms-taglib`instead of `opennms-taglib`. My last commits fix that issue. Should compile now.

**Comment by patrick-schweizer** [comment]
@mvrueden thank you!

File: opennms-taglib/pom.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        <version>21.1.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>opennms-taglib</artifactId>
+    <name>OpenNMS Taglib</name>
+    <packaging>jar</packaging>
+    <dependencies>
+        
+        <dependency><!-- for SimpleTag -->
+            <groupId>org.mortbay.jasper</groupId>
```
[the remark this answers is not in the document]
Comment (patrick-schweizer) [reply opennms-taglib/pom.xml]: It worked without adding a new dependency in opennms-webapp. If I understand correctly we merge ncs-alarm-gui at runtime with opennms-webapp. So I guess yes we can set the scope to provided.

**Pull request merged**